### PR TITLE
xen: cleanup, code quality improvements

### PR DESCRIPTION
**Pull request by SigmaSquadron**
## Description of changes

Minor changes and a cherry-pick of #341429 in preparation for a larger PR that begins building the MiniOS stubdomains.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
- [x] Tested, as applicable:
  I've been dogfooding this branch for a while now.
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 342915"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).
  - Not sure. Let me know if the commits should be squashed.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/virtualization/xen/update.sh [hunk 1]
```diff
@@ -120,13 +127,13 @@ for version in "${supportedVersions[@]}"; do
     echo -e "Found the following patches:\n  \e[1;32mXen\e[0m:     \e[1;33m$discoveredXenPatchesEcho\e[0m\n  \e[1;36mQEMU\e[0m:    \e[1;33m$discoveredQEMUPatchesEcho\e[0m\n  \e[1;36mSeaBIOS\e[0m: \e[1;33m$discoveredSeaBIOSPatchesEcho\e[0m\n  \e[1;36mOVMF\e[0m:    \e[1;33m$discoveredOVMFPatchesEcho\e[0m\n  \e[1;36miPXE\e[0m:    \e[1;33m$discoveredIPXEPatchesEcho\e[0m"
 
     # Prepare patches that are called in ./patches.nix.
-    defaultPatchListInit=("QUBES_REPRODUCIBLE_BUILDS" "XSA_458" "XSA_460" "XSA_461" )
+    defaultPatchListInit=("QUBES_REPRODUCIBLE_BUILDS" "XSA_460" "XSA_461" )
```
Comment (emilazy) [inline pkgs/applications/virtualization/xen/update.sh:130]: What’s the deal with `XSA_458`?

**Review by emilazy — COMMENTED** [review]
Generally LGTM.

File: pkgs/applications/virtualization/xen/generic/0000-xen-ipxe-src-generic.patch [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Hack to make etherboot use pre-fetched iPXE.
 
-diff --git a/tools/firmware/etherboot/Makefile b/tools/firmware/etherboot/Makefile
+diff a/tools/firmware/etherboot/Makefile b/tools/firmware/etherboot/Makefile
```
Comment (emilazy) [inline pkgs/applications/virtualization/xen/generic/0000-xen-ipxe-src-generic.patch:3]: What happened here? Looks like a Git format diff.

Comment (SigmaSquadron) [reply pkgs/applications/virtualization/xen/generic/0000-xen-ipxe-src-generic.patch:3]: Just cosmetics. The flag doesn't really matter one way or the other, so I thought I'd just keep the path. Another patch by Lach had other flags that I also removed for consistency, so I standardised this one as well.

File: pkgs/applications/virtualization/xen/update.sh (hunk 1 above)
Comment (SigmaSquadron) [reply pkgs/applications/virtualization/xen/update.sh:130]: Already included in 4.17.5, 4.18.3 and 4.19.0, so it's unused.

**Comment by SigmaSquadron** [comment]
OfBorg is failing :(

**Comment by SigmaSquadron** [comment]
Big surprise. I was stupid and forgot how newlines work.

**Review by emilazy — APPROVED** [review]

**Comment by SigmaSquadron** [comment]
Part II: #343111

**Pull request merged**